### PR TITLE
add doc and struct of JS code

### PR DESCRIPTION
**Pull request by emugnier**
# Description

This PR adds a very basic documentation of some important part of our code.
You can generate it using `npm run doc`

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

# How Has This Been Tested?

Generated the documentation

### CODE REVIEW

File: src/renderer/timer.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/**
+ * Starts the timer and create the session accordingly
+ * (UI changes are also handle there)
```
Comment (fedarko) [inline src/renderer/timer.js:3]:
```suggestion
 * Starts the timer and creates the session accordingly
 * (UI changes are also handled there)
```

File: src/renderer/timer.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * @param {string} name - Name of the session
+ * @returns {boolean} - True if success else False
```
Comment (fedarko) [inline src/renderer/timer.js:6]: It isn't clear to me how these functions could fail. Could you add a brief description, at least for `startTimer()`?

File: src/storage/Settings.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+   * Store the settings in persistent data.
+   * @return {boolean} True if success else False
+   */
+  store() {
+    // ...
+  }
+
+  /**
+   * Get the settings in persistent data.
+   * @return {boolean} True if success else False
```
Comment (fedarko) [inline src/storage/Settings.js]: Does this function actually load the settings from a file, or does it just return its data to the user? It's unclear to me

File: src/storage/Session.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  // ...
+  }
+
+  /**
+   * Get the session in persistent data.
+   * @return {number} id of the session if success.
+   */
+  getFromLocalStorage() {
+    // ..
+  }
```
Comment (fedarko) [inline src/storage/Session.js]: Unclear what this does - not sure if it like exports the session state or what. Could you elaborate in the docs a bit?

**Review by fedarko — CHANGES_REQUESTED** [review]
Thanks so much for this! I have a few questions about some of the API methods, but for this looks fine.

File: src/storage/Session.js (hunk 4 above)
Comment (emugnier) [reply src/storage/Session.js]: I just removed this function as I am not sure of its usefulness.

File: src/storage/Settings.js (hunk 3 above)
Comment (emugnier) [reply src/storage/Settings.js]: Done

File: src/renderer/timer.js (hunk 2 above)
Comment (emugnier) [reply src/renderer/timer.js:6]: Done

**Review by fedarko — APPROVED** [review]

**Pull request merged**